utilities/infra.py

```
@@ -1140,3 +1148,77 @@ def switch_user_context(context_name: str) -> Generator[None, None, None]:
+    oc_console_cli_download_link = get_oc_console_cli_download_link()
+    LOGGER.info(f"Downloading archive using: url={oc_console_cli_download_link}")
+    urllib3.disable_warnings()
+    local_file_name = os.path.join(tmpdir, oc_console_cli_download_link.split("/")[-1])
+    with requests.get(oc_console_cli_download_link, verify=False, stream=True) as created_request:
+        created_request.raise_for_status()
+        with open(local_file_name, "wb") as file_downloaded:
+            for chunk in created_request.iter_content(chunk_size=8192):
+                file_downloaded.write(chunk)
+    LOGGER.info("Extract the downloaded archive.")
+    if oc_console_cli_download_link.endswith(".zip"):
+        zip_file = zipfile.ZipFile(file=local_file_name)
```

Pylint (3.3.7)

[refactor] Consider using 'with' for resource-allocating operations

(R1732)